src/Aspire.Hosting.Azure.CosmosDB/AzureCosmosDBExtensions.cs

```diff
@@ -229,4 +253,28 @@ public static IResourceBuilder<AzureCosmosDBResource> AddDatabase(this IResource
         builder.Resource.Databases.Add(databaseName);
         return builder;
     }
+
+    /// <summary>
+    /// Configures the Azure Cosmos DB preview emulator to expose the Data Explorer endpoint.
+    /// </summary>
+    /// <param name="builder">Builder for the Cosmos emulator container</param>
+    /// <param name="port">Optional host port to bind the Data Explorer to.</param>
+    /// <returns>Cosmos emulator resource builder.</returns>
+    /// <remarks>
+    /// The Data Explorer is only available with <see cref="RunAsPreviewEmulator"/>.
+    /// </remarks>
+    public static IResourceBuilder<AzureCosmosDBEmulatorResource> WithDataExplorer(this IResourceBuilder<AzureCosmosDBEmulatorResource> builder, int? port = null)
+    {
+        if (!builder.Resource.InnerResource.IsPreviewEmulator)
```

Comment:
Why?

Reply:
Because it doesn't work in the `latest` emulator. See https://github.com/Azure/azure-cosmos-db-emulator-docker/issues/135.